### PR TITLE
perf(insights): aggregate archive series before crossing the Tauri IPC boundary

### LINKED ISSUE #1983: perf(insights): aggregate archive series before crossing the Tauri IPC boundary
## Summary

Move archive bucketing/downsampling for Insights from the React frontend to the Rust/Core + SQLite side, so the frontend receives only the data points required for rendering.

## Current behavior

`useInsightChart.ts` currently:

1. requests all archive records for the selected period,
2. receives the raw records through the Tauri IPC boundary,
3. parses timestamps and groups records into buckets in TypeScript,
4. applies `avg` / `max` / `min`,
5. fills missing buckets,
6. converts the result into chart data.

`useSnapshot.ts` performs a similar operation. It fetches archive records, calculates a step from the requested period, groups data into up to 100 buckets, aggregates each bucket, and fills missing points on the frontend.

The Rust archive query layer currently returns the matching raw rows ordered by timestamp. This means long periods can transfer substantially more records across IPC than the chart ultimately needs.

## Proposal

Add a Core-owned archive-series API that performs bucketing/downsampling before returning data to the frontend.

Conceptually:

```text
SQLite
  -> bucket / aggregate / fill series in Core
  -> compact archive series DTO
  -> Tauri IPC
  -> React / chart rendering
```

The API should support at least:

- CPU / RAM archive series
- GPU archive series
- `avg`, `max`, and `min`
- caller-specified bucket width or desired bucket count
- explicit missing buckets as `null`

For example, a command/API could accept a time range plus a bucket size or desired number of points and return something equivalent to:

```ts
Array<{
  timestamp: number;
  value: number | null;
}>
```

The exact implementation may use SQLite aggregation where practical rather than fetching raw rows and aggregating them in Rust.

## Scope

- Replace the bucketing/aggregation logic in `useInsightChart.ts` with the Core-owned series API.
- Reuse the same aggregation path from Snapshot where possible instead of maintaining a separate frontend implementation.
- Keep presentation concerns in the frontend, including localized date formatting and display-unit formatting such as Celsius/Fahrenheit.
- Preserve the existing archive semantics and chart output.

## Expected benefits

- Reduce the amount of archive data serialized across the Tauri IPC boundary.
- Reduce JavaScript allocations and timestamp parsing for long Insights periods.
- Avoid duplicating archive aggregation logic between Insights and Snapshot.
- Keep data processing close to the SQLite persistence layer that owns the archive data.

## Acceptance criteria

- [ ] Insights no longer buckets raw archive rows in `useInsightChart.ts`.
- [ ] Snapshot uses the Core-owned aggregation path where applicable.
- [ ] `avg`, `max`, and `min` results remain equivalent to the current implementation.
- [ ] Missing intervals remain represented as gaps rather than zero values.
- [ ] Existing supported archive periods continue to render correctly.
- [ ] Add Rust-side tests for bucket boundaries, aggregation modes, null values, and missing buckets.
- [ ] Add/update frontend tests to verify the returned series is rendered without additional aggregation.
- [ ] Compare IPC payload size and Insights rendering/processing cost before and after the change, especially for long periods such as 30 days.

### PR DESCRIPTION
**Pull request by shm11C3**
## Summary

- aggregate CPU, RAM, and GPU archive series in SQLite/Core before returning data through Tauri IPC
- preserve `avg`, `max`, `min`, bucket boundaries, null readings, and explicit missing-bucket gaps
- migrate Insights and Snapshot to render the compact Core-owned series while keeping date and display-unit formatting in the frontend
- regenerate the typed Tauri bindings and update frontend/E2E fixtures for the new series contract

## Related Issues

Closes #1983

## Type of Change

- [ ] Bug fix (`fix/` branch)
- [ ] New feature (`feat/` branch)
- [ ] Refactoring (`refactor/` branch)
- [x] Performance (`perf/` branch)
- [ ] Documentation (`docs/` branch)
- [ ] Dependencies update
- [ ] Other (`chore/` branch)

## Performance Comparison

A synthetic 30-day, one-minute archive comparison using 12-hour output buckets produced:

- points crossing the boundary: 43,201 -> 61
- serialized JSON payload: 3,366,098 bytes -> 3,293 bytes (99.90% reduction)
- frontend processing median over 100 runs: 8.653 ms -> 0.004 ms

This comparison measures synthetic payload and frontend processing cost in Node.js. It does not include real-device SQLite query time or UI rendering performance.

## Screenshots / Videos

Not applicable. The rendered chart contract is preserved; this change moves archive processing ownership behind IPC.

## Test Plan

- [x] `cargo test -p hardviz-core` (261 unit tests and 2 integration tests)
- [x] `cargo test -p hardware_visualizer --lib` (239 tests)
- [x] `cargo clippy -p hardviz-core --all-targets -- -D warnings`
- [x] `cargo clippy -p hardware_visualizer --all-targets -- -D warnings`
- [x] `cargo fmt --all -- --check`
- [x] `npm test` (606 tests)
- [x] `npm run lint:ci`
- [x] `npm run build`
- [x] regenerate `src/rspc/bindings.ts` with the bindings export binary
- [x] `git diff --check`

## Checklist

- [x] Self-reviewed the code
- [x] Linting and formatting pass (`npm run lint && npm run format` / `cargo tauri-lint && cargo tauri-fmt`)
- [x] Tests pass (`npm test` / `cargo tauri-test`)
- [x] No new warnings or errors

## Summary by CodeRabbit

- **New Features**
  - Added time-bucketed archive series for CPU, memory, and GPU metrics.
  - Supports configurable bucket intervals and timestamp alignment.
  - Provides average, minimum, and maximum aggregation options.
  - Preserves empty intervals as null values for clearer chart timelines.
  - Validates time ranges, bucket sizes, and result limits.

- **Improvements**
  - Hardware insight and snapshot charts now use pre-aggregated archive series for more consistent data and rendering.

### CODE REVIEW

File: core/src/infrastructure/database/archive_queries.rs [hunk 1]
```diff
@@ -104,36 +228,255 @@ struct GpuNameRow {
...
+      return Err(ArchiveSeriesError::InvalidTimeRange);
+    }
+
+    let (first, last) = match bucket_timestamp {
+      ArchiveBucketTimestamp::Start => {
+        (floor_to_bucket(start, width), floor_to_bucket(end, width))
+      }
+      ArchiveBucketTimestamp::End => (
+        ceil_to_bucket(start.saturating_sub(ARCHIVE_INTERVAL_MILLISECONDS), width),
+        ceil_to_bucket(end, width),
```
[the remark this answers is not in the document]
Comment (shm11C3) [reply core/src/infrastructure/database/archive_queries.rs]: Addressed in 57d249ed. End-aligned bounds now use ceil(query_start / bucket_width) directly, matching the previous frontend calculation. The updated Rust regression starts at 00:00:00.001 and asserts that the first returned point is 00:01:00, with no extra leading null bucket.

File: core/src/infrastructure/database/archive_queries.rs [hunk 2]
```diff
@@ -104,36 +228,255 @@ struct GpuNameRow {
...
+      series.push(ArchiveSeriesPoint {
+        timestamp,
+        value: row.and_then(|row| row.value),
+      });
+    }
+
+    timestamp += bounds.width;
+  }
+
+  series
```
[the remark this answers is not in the document]
Comment (shm11C3) [reply core/src/infrastructure/database/archive_queries.rs:490]: Declined after verification: the existing Rust test already covers this contract. Its end is 00:02:00.001, so the computed trailing boundary is 00:03:00, but no row maps there; the assertion ends at 00:02:00 and therefore proves omission of the empty partial bucket. buildArchiveSeries models an already-populated Core response rather than raw archive rows, so emitting its final populated point is consistent.
